### ISSUE TITLE
[no-unused-vars] (v4-alpha.1) False positive when in an interface

### ISSUE THREAD

**Issue by eamodio** [typescript-eslint/typescript-eslint#2331]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
		"warn",
		{
			"args": "after-used",
			"argsIgnorePattern": "^_",
			"ignoreRestSiblings": true,
			"varsIgnorePattern": "^_$"
		}
	]
  }
}
```

```TS
export interface Event<T> {
	(listener: (e: T) => any, thisArgs?: any, disposables?: Disposable[]): Disposable;
}
```

**Expected Result**

No errors

**Actual Result**

```
  11:3   warning  'listener' is defined but never used. Allowed unused args must match /^_/u     @typescript-eslint/no-unused-vars
  11:28  warning  'thisArgs' is defined but never used. Allowed unused args must match /^_/u     @typescript-eslint/no-unused-vars
  11:44  warning  'disposables' is defined but never used. Allowed unused args must match /^_/u  @typescript-eslint/no-unused-vars
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.0-alpha.1` |
| `@typescript-eslint/parser`        | `4.0.0-alpha.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `12.16.2` |
| `npm`                              | `6.14.4` |
